test/memory.test.js

```diff
@@ -463,25 +463,88 @@ describe('Memory connector', function() {
       });
     });
 
+    it('automigrate all models - promise variant', function(done) {
+      ds.automigrate()
+        .then(function(result) {
+          done();
+        })
+        .catch(function(err){
+          done(err);
+        });
+    });
+
     it('automigrate one model', function(done) {
       ds.automigrate('m1', function(err) {
         done(err);
       });
     });
 
+    it('automigrate one model - promise variant', function(done) {
+      ds.automigrate('m1')
+        .then(function(result) {
+          done();
+        })
+        .catch(function(err){
+          done(err);
+        });
+    });
+
     it('automigrate one or more models in an array', function(done) {
       ds.automigrate(['m1'], function(err) {
         done(err);
       });
     });
 
+    it('automigrate one or more models in an array - promise variant', function(done) {
+      ds.automigrate(['m1'])
+        .then(function(result) {
+          done();
+        })
+        .catch(function(err){
+          done(err);
+        });
+    });
+
     it('automigrate reports errors for models not attached', function(done) {
       ds.automigrate(['m1', 'm2'], function(err) {
         err.should.be.an.instanceOf(Error);
         done();
       });
     });
+
+    it('automigrate reports errors for models not attached - promise variant', function(done) {
+      ds.automigrate(['m1', 'm2'])
+        .then(done)
+        .catch(function(err){
+          err.should.be.an.instanceOf(Error);
+          done();
+        });
+    });
+
+  });
+
+  it('automigrate does NOT report error when NO models are attached', function(done) {
+    var ds;
+    ds = new DataSource({
+        connector: 'memory'
+      });
```

Comment:
I am proposing to create a `describe` block and move the initialisation of an empty datasource to a `beforeEach` hook, in order to reduce the amount of duplicated code.

```js
describe('automigrate when NO models are attached', function() {\
  var ds;
  beforeEach(/*setup ds */);
  it('does NOT report error', /*...*/);
  it('does NOT report error - promise variant');
});
```

Thoughts?